src/main/java/javax/validation/TypeParameter.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Bean Validation API
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package javax.validation;
+
+import java.io.Serializable;
+import java.lang.reflect.TypeVariable;
+
+/**
+ * Serializable structure representing a type parameter in a generic declaration.
+ * <p>
+ * This is a serializable and very simplified replacement for {@link TypeVariable}.
+ * <p>
+ * For now, it only supports class level generic declaration.
+ *
+ * @author Guillaume Smet
+ * @since 2.0
+ */
+public class TypeParameter implements Serializable {
+
+	private final String name;
+
+	private final int index;
+
+	private GenericDeclaration genericDeclaration;
+
+	public static TypeParameter of(TypeVariable<?> typeVariable) {
+		java.lang.reflect.GenericDeclaration originalGenericDeclaration = (java.lang.reflect.GenericDeclaration) typeVariable.getGenericDeclaration();
+		if ( !( originalGenericDeclaration instanceof Class ) ) {
+			throw new ValidationException( "Only class level type variables are supported." );
+		}
+
+		GenericDeclaration genericDeclaration = new ClassGenericDeclaration( (Class<?>) originalGenericDeclaration );
+		TypeParameter typeParameter = null;
+
+		for ( TypeParameter currentTypeParameter : genericDeclaration.getTypeParameters() ) {
+			currentTypeParameter.genericDeclaration = genericDeclaration;
+			if ( typeVariable.getName().equals( currentTypeParameter.getName() ) ) {
+				typeParameter = currentTypeParameter;
+			}
+		}
+
+		return typeParameter;
+	}
+
+	private TypeParameter(String name, int index) {
+		this.name = name;
+		this.index = index;
+	}
+
+	/**
+	 * Returns the name of the type parameter.
+	 *
+	 * @return the name of the type parameter
+	 */
+	public String getName() {
+		return name;
+	}
+
+	/**
+	 * Returns the index of the type parameter in the array of type parameters of the generic declaration.
+	 *
+	 * @return the index of the type parameter
+	 */
+	public int getIndex() {
+		return index;
+	}
+
+	/**
+	 * Returns the generic declaration defining this type parameter.
+	 *
+	 * @return the generic declaration
+	 */
+	public GenericDeclaration getGenericDeclaration() {
```

Comment:
I think it's useful information coming from `TypeVariable` to know what we're talking about. Just having the name seems very limited to me.

Btw, it's used in HV to generate a conservative `toString()` for now - we'll see if we keep that. See https://github.com/hibernate/hibernate-validator/pull/683/files#diff-ec06ef14de55635f374c64ab50ac8405R393 . It might be moved into the implementation if you prefer but I think we would lose too much information.